### PR TITLE
DOC: Ignore attribute FutureWarning in doc build

### LINKED ISSUE #22987: DOC: building documentation emits many FutureWarnings
There are many FutureWarnings when [building docs](https://app.circleci.com/pipelines/github/numpy/numpy/17525/workflows/a1ca64ae-8bb7-457f-97a1-880b0672944d/jobs/29548?invite=true#step-109-152) of the form
> `/home/circleci/repo/venv/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py:678: FutureWarning: In the future `np.bool` will be defined as the corresponding NumPy scalar.`
>  `return getattr(mod, name_parts[-1]), mod, modname`
> `/home/circleci/repo/venv/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py:701: FutureWarning: In the future `np.bool` will be defined as the corresponding NumPy scalar.`
>  `obj = getattr(obj, obj_name)`
> `/home/circleci/repo/venv/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py:678: FutureWarning: In the future `np.bytes` will be defined as the corresponding NumPy scalar.`
>  `return getattr(mod, name_parts[-1]), mod, modname`
> `/home/circleci/repo/venv/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py:701: FutureWarning: In the future `np.bytes` will be defined as the corresponding NumPy scalar.`
>  `obj = getattr(obj, obj_name)`
> `/home/circleci/repo/venv/lib/python3.9/site-packages/sphinx/ext/autosummary/__init__.py:678: FutureWarning: In the future `np.str` will be defined as the corresponding NumPy scalar.`
> ` return getattr(mod, name_parts[-1]), mod, modname`

I am also seeing an error when building locally with CPython 3.9.16:
```
WARNING: [autosummary] failed to import numpy.finfo.machar.
Possible hints:
* AttributeError: type object 'finfo' has no attribute 'machar'
* ImportError: 
* ModuleNotFoundError: No module named 'numpy.finfo'
```

but I don't see it in CI. Indeed, I cannot run `python -c "import numpy.finfo"`, only `python -c "from numpy import finfo"`. 
Here is the autodoc line: https://github.com/numpy/numpy/blob/df1ee34c7faea1cfc2aef0719444b6fa43887b42/doc/source/reference/routines.dtype.rst?plain=1#L28-L32

### PR DESCRIPTION
**Pull request by seberg**
The docs somehow seem to access `numpy.str`, etc. during the autodoc generation probably.
This adds a specific warning filter to hide it.

Closes gh-22987

### CODE REVIEW

**Comment by rossbar** [comment]
I was looking at this issue a bit - I think this is baked into how `autosummary` does object discovery:

https://github.com/sphinx-doc/sphinx/blob/e17f39e7b535f4849754d46f2cd2e411df8aa535/sphinx/ext/autosummary/__init__.py#L680-L692

Of course, `numpydoc` hooks custom `mangle_docstring` and `mangle_signature` functions into the autodoc process, so they may be contributing as well. I spent a bit of time trying to reconstruct the call pattern *outside* of sphinx_build to try to develop a more minimal example for reproducing the warning, but I wasn't able to do so easily. 

Long story short - I support suppressing the warning(s) as I don't think there's a straightforward way to avoid it without digging more deeply into sphinx+extensions machinery.

**Comment by mattip** [comment]
Thanks @seberg
